crates/re_space_view_time_series/src/space_view_class.rs

```diff
@@ -621,72 +624,37 @@ impl SpaceViewClass for TimeSeriesSpaceView {
-    let corner = corner.unwrap_or(DEFAULT_LEGEND_CORNER.into());
+    let component_names = [Visible::name(), Corner2D::name()];
+    let component_results =
+        blueprint_db.latest_at(blueprint_query, &blueprint_path, component_names);
 
     let sub_prop_ui = |re_ui: &re_ui::ReUi, ui: &mut egui::Ui| {
-        // TODO(ab): components should provide the `value_fn` closure or the entire
-        // `PropertyContent` object
-
-        //
-        // Visible
-        //
-
-        let mut edit_visibility = visible;
-        list_item2::ListItem::new(re_ui)
-            .interactive(false)
-            .show_flat(
-                ui,
-                list_item2::PropertyContent::new("Visible").value_bool_mut(&mut edit_visibility.0),
-            );
-        if visible != edit_visibility {
-            ctx.save_blueprint_component(&blueprint_path, &edit_visibility);
-        }
-
-        //
-        // Corner
-        //
-
-        let mut edit_corner = corner;
-        list_item2::ListItem::new(re_ui)
-            .interactive(false)
-            .show_flat(
-                ui,
-                list_item2::PropertyContent::new("Corner").value_fn(|_, ui, _| {
-                    egui::ComboBox::from_id_source("legend_corner")
-                        .selected_text(format!("{corner}"))
-                        .show_ui(ui, |ui| {
-                            ui.style_mut().wrap = Some(false);
-                            ui.set_min_width(64.0);
-
-                            ui.selectable_value(
-                                &mut edit_corner,
-                                egui_plot::Corner::LeftTop.into(),
-                                format!("{}", Corner2D::from(egui_plot::Corner::LeftTop)),
-                            );
-                            ui.selectable_value(
-                                &mut edit_corner,
-                                egui_plot::Corner::RightTop.into(),
-                                format!("{}", Corner2D::from(egui_plot::Corner::RightTop)),
-                            );
-                            ui.selectable_value(
-                                &mut edit_corner,
-                                egui_plot::Corner::LeftBottom.into(),
-                                format!("{}", Corner2D::from(egui_plot::Corner::LeftBottom)),
-                            );
-                            ui.selectable_value(
-                                &mut edit_corner,
-                                egui_plot::Corner::RightBottom.into(),
-                                format!("{}", Corner2D::from(egui_plot::Corner::RightBottom)),
+        for component_name in component_names {
+            list_item2::ListItem::new(re_ui)
+                .interactive(false)
+                .show_flat(
+                    ui,
+                    // TODO(andreas): Note that we loose the archetype's field name here, instead we label the item with the component name.
```

Comment:
Somewhat related, we'd need to route some textual information to hover tooltips.